xlators/cluster/afr/src/afr-open.c

```diff
@@ -271,11 +274,204 @@ afr_fd_ctx_need_open(fd_t *fd, xlator_t *this, unsigned char *need_open)
     return count;
 }
 
+static int
+afr_do_fix_open(call_frame_t *frame, xlator_t *this)
+{
+    afr_local_t *local = frame->local;
+    afr_private_t *priv = NULL;
+    unsigned char *need_open = NULL;
+    int i = 0;
+    int need_open_count = 0;
+
+    priv = this->private;
+
+    need_open = alloca0(priv->child_count);
+
+    need_open_count = afr_fd_ctx_need_open(local->fd, this, need_open);
+    if (!need_open_count) {
+        goto out;
+    }
+    gf_msg_debug(this->name, 0, "need open count: %d", need_open_count);
+    local->call_count = need_open_count;
+
+    for (i = 0; i < priv->child_count; i++) {
+        if (!need_open[i])
+            continue;
+
+        LOCK(&local->fd->lock);
+        {
+            local->fd_ctx->opened_on[i] = AFR_FD_OPENING;
+        }
+        UNLOCK(&local->fd->lock);
```

Comment:
Created 2 functions now, one to get the need_open count in afr_fix_open() where we do not change the opened_on[] status and another to get the current status of opened_on[]  and if it is set to NOT_OPENED then  only will reset it to OPENING to avoid this.